### PR TITLE
Added Annotation config.kubernetes.io/depends-on

### PR DESCRIPTION
**Pull request by AryanSharma9917**
Documented `config.kubernetes.io/depends-on` annotation.
This PR fixes: #42150

### CODE REVIEW

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 1]
```diff
@@ -262,6 +262,22 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
...
+### config.kubernetes.io/depends-on
+
+Type: Annotation
+
+Example: `config.kubernetes.io/depends-on: "OBJECT_REFERENCE"` 
+
+Used on: All objects
+
+This annotation is used on the target object to be modified to configure [Depends-on](https://kpt.dev/reference/annotations/depends-on/). This annotation specifies one or more resource dependencies.
+The annotation value accepts a list of resource references, delimited by commas.
```
Comment (Ritikaa96) [inline content/en/docs/reference/labels-annotations-taints/_index.md:274]:
``` suggestion
This annotation is used on the target object to be modified to configure explicit dependency ordering. 
The annotation helps to determine the apply order, dependency reconciliation while applying and deletion and prevents accidental premature deletion.
See [Depends-on](https://kpt.dev/reference/annotations/depends-on/) for more information. 
The annotation value accepts a list of resource references, delimited by commas.
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 2]
```diff
@@ -262,6 +262,22 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
...
+
+**Resource reference**
+- **Group**: This represents the API group of the resource (e.g., apps, v1, extensions).
+- **Kind**: This indicates the type of resource (e.g., Pod, Service, Deployment).
+- **Name**: The name is the unique identifier of the specific resource instance.
```
Comment (Ritikaa96) [inline content/en/docs/reference/labels-annotations-taints/_index.md:279]:
```suggestion
For eg. If `config.kubernetes.io/depends-on: /namespaces/default/Pod/pod-a` annotation is added to `pod-b`,  it identifies that pod-a must be successfully applied prior to pod-a actuation.
```
I think the link will explain what are group, kind, and name. Also, they weren't mentioned in the example and seem like popped up out of nowhere.

Comment (Ritikaa96) [reply content/en/docs/reference/labels-annotations-taints/_index.md:279]: Please be noted the eg is already available on the linked page. However, we can add it in place of the Example section. If it seems repetitive I think the present Example section works fine as well, but less informative. Just feel like the explanation helps understanding annotation better.
WDYT, @sftim ?  need a second opinion.

**Review by Ritikaa96 — CHANGES_REQUESTED** [review]

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 2 above)
Comment (AryanSharma9917) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: Hey @Ritikaa96, I have made the changes could you please give it a look

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 3]
```diff
@@ -262,6 +262,22 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
...
+### config.kubernetes.io/depends-on
+
+Type: Annotation
+
+Example: `config.kubernetes.io/depends-on: "OBJECT_REFERENCE"` 
+
+Used on: All objects
+
+This annotation is used on the target object to be modified to configure explicit dependency ordering. 
+
```
Comment (Ritikaa96) [inline content/en/docs/reference/labels-annotations-taints/_index.md:274]:
```suggestion
```

**Review by Ritikaa96 — CHANGES_REQUESTED** [review]

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 3 above)
Comment (Ritikaa96) [reply content/en/docs/reference/labels-annotations-taints/_index.md:274]: Let's remove redundant spaces and lines.

**Comment by Ritikaa96** [comment]
Thanks @AryanSharma9917 
/lgtm

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 4]
```diff
@@ -262,6 +262,21 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
 which is used by Kustomize and similar third-party tools.
 For example, Kustomize removes objects with this annotation from its final build output.
 
+### config.kubernetes.io/depends-on
+
+Type: Annotation
+
+Example: `config.kubernetes.io/depends-on: "OBJECT_REFERENCE"`
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:269]: This doesn't look right - is that a valid actual example?

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 5]
```diff
@@ -262,6 +262,21 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
...
+Example: `config.kubernetes.io/depends-on: "OBJECT_REFERENCE"` 
+
+Used on: All objects
+
+This annotation is used on the target object to be modified to configure explicit dependency ordering. 
+The annotation helps to determine the apply order, and dependency reconciliation while applying and deletion and prevents accidental premature deletion.
+
+For eg. If `config.kubernetes.io/depends-on: /namespaces/default/Pod/pod-a` annotation is added to `pod-b`,  it identifies that pod-a must be successfully applied prior to pod-a actuation.
+
+See [Depends-on](https://kpt.dev/reference/annotations/depends-on/) for more information.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:278]:
```suggestion
For more details, see [depends on](https://kpt.dev/reference/annotations/depends-on/) in the KPT
documentation.
```

**Review by sftim — COMMENTED** [review]
/lgtm cancel

I don't think the example is valid.

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 4 above)
Comment (AryanSharma9917) [reply content/en/docs/reference/labels-annotations-taints/_index.md:269]: @sftim should I mention this in the example value?
```
annotations:
  config.kubernetes.io/depends-on: |
    /namespaces/default/Pod/pod-a,
    /namespaces/default/Pod/pod-c

```

**Comment by AryanSharma9917** [comment]
Hey @sftim is my PR ready to be merged?

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 6]
```diff
@@ -262,6 +262,29 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
...
+```
+
+Used on: All objects
+
+This annotation is used on the target object to be modified to configure explicit dependency ordering. 
+The annotation helps to determine the apply order, and dependency reconciliation while applying and deletion and prevents accidental premature deletion.
+
+For example If `config.kubernetes.io/depends-on: /namespaces/default/Pod/pod-a` annotation is added to `pod-b`,  it identifies that pod-a must be successfully applied prior to pod-a actuation.
+
+For more details, see [depends on](https://kpt.dev/reference/annotations/depends-on/) in the KPT documentation.
```
Comment (zevisert) [inline content/en/docs/reference/labels-annotations-taints/_index.md:286]: I know I also first found this from KPT, but I wonder if a better permalink is https://github.com/kubernetes-sigs/cli-utils/blob/0b156cb0425fdb29a436d13f840a39039558c10e/README.md#explicit-dependency-ordering

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 7]
```diff
@@ -262,6 +262,29 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
...
+
+Example: `config.kubernetes.io/depends-on:`
+
+```yaml
+  annotations:
+  config.kubernetes.io/depends-on: |
+    /namespaces/default/Pod/pod-a,
+    /namespaces/default/Pod/pod-c
+
+```
```
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:277]: This example is not correct.
The value of an annotation has to be a string.

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 8]
```diff
@@ -262,6 +262,29 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
...
+
+Used on: All objects
+
+This annotation is used on the target object to be modified to configure explicit dependency ordering. 
+The annotation helps to determine the apply order, and dependency reconciliation while applying and deletion and prevents accidental premature deletion.
```
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:282]: please wrap the long lines as we do for other paragraphs.

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 9]
```diff
@@ -262,6 +262,29 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
...
+
+For example If `config.kubernetes.io/depends-on: /namespaces/default/Pod/pod-a` annotation is added to `pod-b`,  it identifies that pod-a must be successfully applied prior to pod-a actuation.
```
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:284]: Please be careful with capitals, spaces, and typos.

```suggestion
For example, if the annotation `config.kubernetes.io/depends-on: /namespaces/default/Pod/pod-a`
is added to a Pod `pod-b`, it identifies that `pod-a` must be successfully applied
prior to `pod-b` actuation.
```

**Review by tengqm — COMMENTED** [review]
Overall, I don't think we want to add this annotation to k8s docs.

**Comment by sftim** [comment]
> Overall, I don't think we want to add this annotation to k8s docs.

We should document it; maybe _this_ PR is not yet ready to merge though?

**Comment by tengqm** [comment]
> > Overall, I don't think we want to add this annotation to k8s docs.
> 
> We should document it; maybe _this_ PR is not yet ready to merge though?

I'm not 100% sure whether this annotation will be there if I'm using a vanilla Kubernetes.
I don't have an objection if the answer is yes.

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 10]
```diff
@@ -262,6 +262,27 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
...
+### config.kubernetes.io/depends-on
+
+Type: Annotation
+
+Example: `config.kubernetes.io/depends-on:`
+
+```yaml
+  annotations:
+    config.kubernetes.io/depends-on: "/namespaces/default/Pod/pod-a,/namespaces/default/Pod/pod-c"
+```
```
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:274]:
```suggestion
Example: `config.kubernetes.io/depends-on:"/namespaces/default/Pod/pod-a,/namespaces/default/Pod/pod-c"`
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 11]
```diff
@@ -262,6 +262,27 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
...
+
+Used on: All objects
+
+This annotation is used on the target object to be modified to configure explicit 
+dependency ordering. The annotation helps to determine the apply order, and dependency 
+reconciliation while applying and deletion and prevents accidental premature deletion.
+
+For example If the annotation `config.kubernetes.io/depends-on: /namespaces/default/Pod/pod-a` is added to `pod-b`,  it identifies that `pod`-a must be successfully applied prior to `pod-b` actuation.
```
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:282]:
```suggestion
For example If the annotation `config.kubernetes.io/depends-on: /namespaces/default/Pod/pod-a`
is added to `pod-b`, it identifies that `pod-a` must be successfully applied
prior to `pod-b` actuation.
```

Comment (tengqm) [reply content/en/docs/reference/labels-annotations-taints/_index.md:282]: We'd better document which tool is attaching this annotation.
A user needs to know when this appears and how to interpret it.

**Comment by tengqm** [comment]
Better state where this annotation comes from, like "this annotation is added to a Pod by blahblah", users know when to check it and when not to check it.

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 12]
```diff
@@ -299,6 +299,7 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
 which is used by Kustomize and similar third-party tools.
 For example, Kustomize removes objects with this annotation from its final build output.
 
+depends-on
```
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:302]: what's this?

Comment (sftim) [reply content/en/docs/reference/labels-annotations-taints/_index.md:302]:
```suggestion
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 13]
```diff
@@ -299,6 +299,25 @@ This annotation is part of the Kubernetes Resource Model (KRM) Functions Specifi
...
+### config.kubernetes.io/depends-on
+
+Type: Annotation
+
+Example: `config.kubernetes.io/depends-on:"/namespaces/default/Pod/pod-a,/namespaces/default/Pod/pod-c"`
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:307]:
```suggestion
Example: `config.kubernetes.io/depends-on: "/namespaces/default/Pod/pod-a,/namespaces/default/Pod/pod-c"`
```

**Comment by sftim** [comment]
@AryanSharma9917 would you be willing to update this pull request based on the feedback so far?

**Comment by Ritikaa96** [comment]
Hi @AryanSharma9917,  since there has been no response/commits despite reminders for quite some time, I'll be closing the PR. If you find the bandwidth to work on it again, please reopen it & address the suggestions provided by the PR reviewers.

**Pull request closed without merging**